docs/tech-notes/sql.md

```diff
@@ -442,197 +502,44 @@ Two things are involved here:
 
 **Role:** turn the AST into a logical plan.
 
-**Interface:**  `prepare()` and `newPlan()` in `sql/plan.go`.
+**Interface:** see `opt/optbuilder`.
 
 **How:**
 
 - in-order depth-first recursive traversal of the AST;
 - invokes semantics checks on the way;
-- constructs `planNode` instances on the way back from the recursion;
-  - or results in an error if some semantic check fails;
-- the resulting `planNode` tree *is* the logical plan.
-
-Note: some statements are not really translated to a logical plan,
-like `SET`. I call them *"immediate"*. These are executed (ran)
-immediately in `newPlan()` after semantic checking, and result in an
-"empty plan" from the perspective of the code using the logical
-planner. The `prepare()` interface exists so that these statements are
-ignored during SQL prepare, but this in turn also means that a SQL
-prepare operation is unable to run semantic checks on immediate
-statements.
-
-**Code:** the various `planNode` constructors in the `sql` package.
-
-**Whom to ask for details:** like for semantic checks, there's no
-single point of expertise currently able to talk about the commonalities of
-logical plan construction. Some areas of expertise have emerged:
-
-- `scanNode`: radu, andrei;
-- interaction between `renderNode`/`filterNode`/`groupNode`: radu, knz;
-- `windowNode`: nathan;
-- `sortNode`: radu, nathan, knz;
-- `copyNode`: mattj;
-- `updateNode`/`insertNode`/`upsertNode`/`deleteNode`: peter, danhhz, dt;
-- `create*Node`, `alter*Node`: dt, vivek, danhhz;
-- the others: a bit everyone who ever worked in the `sql` package.
+- constructs a memoized tree of "relational expression nodes".
```

Comment:
I think "memoized tree" is not especially evocative or important - I think "tree" would probably be sufficient for the depth of explanation here (it's not even really using the memo too much at this point, since there won't be any exploration yet).